### ISSUE TITLE
Sorry, you cannot edit this document with others

### ISSUE THREAD

**Issue by OrdinaryWorld111** [petrsvihlik/WopiHost#159]
Word and pptx can be edited multiple times by running VS, but after being deployed to the IIS server, Word and pptx can only be edited once. Opening them again will result in an error message: Sorry, you cannot edit this document with others. Excel can be edited multiple times. Have you ever encountered similar problems? Please give me some help

**Comment by petrsvihlik** [comment]
Nope. Have you tried to compare the files before and after editing? You can unzip any open office docs (including PPTXes) and compare the text files. Perhaps you can find what got corrupted.

**Comment by petrsvihlik** [comment]
The symptom (Word/PPTX edits succeed in Visual Studio but fail on the *second* open under IIS, while Excel keeps working) is a classic lock-store lifetime mismatch: the default `MemoryLockProvider` (in-process `ConcurrentDictionary`, 30-min expiry) doesn't survive IIS app-pool recycles, and Excel survives because it uses the Cobalt/co-authoring path which is locked differently. The host is behaving correctly — the in-memory provider just isn't suitable for IIS / multi-instance.

The proper fix is to use a distributed lock provider. [`WopiHost.AzureLockProvider`](https://github.com/petrsvihlik/WopiHost/tree/master/src/WopiHost.AzureLockProvider) ships exactly that. PR #332 adds a callout to the `MemoryLockProvider` README explaining this trap.

Closing — please reopen if you swap to a distributed provider and the symptom persists.
